protocol/operation.rei

```diff
@@ -22,14 +22,6 @@ module Main_chain: {
 
   let make:
     (~tezos_hash: Tezos.Operation_hash.t, ~tezos_index: int, ~kind: kind) => t;
-  let verify:
-    (
-      ~hash: BLAKE2B.t,
-      ~tezos_hash: Tezos.Operation_hash.t,
-      ~tezos_index: int,
-      ~kind: kind
-    ) =>
-    result(t, string);
```

Comment:
This function should not be used, and it's not, that's why I removed it from the interface.

Reply:
Sorry, I'm new here: why shouldn't it be used?

Reply:
It's called in of_yojson, which is the only time you should have to verify

Reply:
@aguillon it's an internal function, the only way to create an operation is by making one or receiving one from the network, so verify should never be called.

This function is not unsound on itself, is just a bad API.